app/views/foreman/unattended/snippets/_subscription_manager_registration.erb

```diff
@@ -1,12 +1,18 @@
 <% if @host.params['kt_activation_keys'] %>
-# add subscription manager
-yum -t -y -e 0 install subscription-manager
-rpm -ivh <%= subscription_manager_configuration_url(@host) %>
+  # add subscription manager
+ <% if @host.operatingsystem.name.match(/.*atomic.*/i) %>
+  curl -s <%= subscription_manager_configuration_url(@host, false) %> | IS_ATOMIC=true bash
```

Comment:
Note this PR also needs this PR https://github.com/Katello/puppet-certs/pull/81 for full functionality. We need to add IS_ATOMIC var during provisioning for katello-rhsm-consumer to realize its on an atomic machine.